### PR TITLE
chore: adds the missing components and updates the current ones

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR aims to align the design system with latest changes on the main app repository

### CODE REVIEW

File: src/components/typography/H3.tsx
Comment (drmarro) [inline src/components/typography/H3.tsx:1]: It seems that you've migrated from the io-app the H2 component into the H3 one.
Is it right ?

File: src/components/typography/H3.tsx [hunk 1]
```diff
@@ -1,101 +1,35 @@
-import React from "react";
```
Comment (drmarro) [inline src/components/typography/H3.tsx:1]: It seems that you've migrated from the io-app the H2 component into the H3 one.
Is it right ?

Comment (CrisTofani) [reply src/components/typography/H3.tsx:1]: Actually not, H2 component has the following attributes:

```
export const h2FontSize = 20;
export const h2LineHeight = 24;
```

File: src/components/typography/Link.tsx [hunk 2]
```diff
@@ -1,49 +1,45 @@
...
-    TypographyProps<AllowedWeight, AllowedColors>
+  TypographyProps<AllowedWeight, AllowedColors>
 > &
-    AllowedFontSize;
+  AllowedFontSize;
+
+const fontName: IOFontFamily = "TitilliumWeb";
+const fontSizeMapping: Record<FontSize, number> = { regular: 16, small: 14 };
+
+export const linkLegacyDefaultColor: AllowedColors = "blue";
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply src/components/typography/Link.tsx]: done into [1079e84](https://github.com/pagopa/io-app-design-system/pull/32/commits/1079e8412f590601d1d6a9a591951e97031852b7)

**Review by drmarro — APPROVED** [review]

**Pull request merged**